core/generated-sources/src/main/kotlin/org/jetbrains/kotlinx/dataframe/impl/schema/Utils.kt

```diff
@@ -141,5 +144,53 @@ internal fun DataFrameSchema.createEmptyDataFrame(numberOfRows: Int): AnyFrame =
 internal fun createEmptyDataFrameOf(clazz: KClass<*>): AnyFrame =
     MarkersExtractor.get(clazz).schema.createEmptyDataFrame()
 
-internal fun getPropertiesOrder(clazz: KClass<*>): Map<String, Int> =
-    clazz.primaryConstructor?.parameters?.mapNotNull { it.name }?.mapIndexed { i, v -> v to i }?.toMap() ?: emptyMap()
+internal fun getPropertyOrderFromPrimaryConstructor(clazz: KClass<*>): Map<String, Int>? =
+    clazz.primaryConstructor
+        ?.parameters
+        ?.mapNotNull { it.name }
+        ?.mapIndexed { i, v -> v to i }
+        ?.toMap()
+
+internal fun getPropertyOrderFromAllConstructors(clazz: KClass<*>): List<Map<String, Int>> =
+    clazz.constructors
+        .map { constructor ->
+            constructor.parameters
+                .mapNotNull { it.name }
+                .mapIndexed { i, v -> v to i }
+                .toMap()
+        }
+
+/**
+ * Sorts [this] according to the order of them in the constructors of [klass].
+ * It prefers the primary constructor if it exists, else it falls back to the other constructors to do the sorting.
+ * Finally, it falls back to lexicographical sorting if a property does not exist in any constructor.
+ */
+internal fun <T> Iterable<KCallable<T>>.sortWithConstructors(klass: KClass<*>): List<KCallable<T>> {
```

Comment:
Honestly i still don't understand what to expect from this as a user. I'd like something simpler

Reply:
nope :/ reflection returns them completely random.

The idea was to come as close to what a user expects the order of columns to be as possible. So if they have a weird class like

```kt
class A {
  constructor()
  constructor(foo: Int, bar: Int)
  constructor(bar: Int, too: Int)
  ...
}
```
well, you expect `foo` to be before `bar` and `too` after it, right? That's what this function achieves. It sorts the columns according to the order of all constructors. If there's a primary constructor, obviously that is used first.

Reply:
Now simplified to just sorting lexicographically and by primary/single constructor